lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -71,81 +71,131 @@ void SetMessageManagerHandler::SetAmfSessionContext(
-     * Code for relase state, then creating
+    /* Read the proto message and check for state. Get the config out of
+     * proto. Code for relase state, then creating
      */
     // Requested message from AMF to release the session
     if (cfg.common_context.sm_session_state() == RELEASED_4) {
-      if (cfg.common_context.sm_session_version() != 0) {
-        MLOG(MERROR) << "Wrong version received from AMF for IMSI " << imsi
-                     << " but continuing release request";
+      if (cfg.common_context.sm_session_version() == 0) {
+        MLOG(MERROR) << "Wrong version received from AMF for " << imsi
+                     << " of PDU id:" << pdu_id;
+        Status status(grpc::OUT_OF_RANGE, "Version number Out of Range");
+        response_callback(status, SmContextVoid());
+        return;
       }
       MLOG(MINFO) << "Release request for session from IMSI: " << imsi
-                  << " DNN " << dnn;
+                  << " pdu_id " << pdu_id;
       /* Read the SessionMap from global session_store,
        * if it is not found, it will be added w.r.t imsi
        */
       auto session_map = session_store_.read_sessions({imsi});
-      initiate_release_session(session_map, dnn, imsi);
+      initiate_release_session(session_map, pdu_id, imsi);
+      response_callback(Status::OK, SmContextVoid());
     } else {
-      // The Event Based main_thread invocation and runs to handle session state
-      std::string session_id = id_gen_.gen_session_id(imsi);
+      // The Event Based main_thread invocation and runs to handle session
+      // state
       MLOG(MINFO) << "Requested session from UE with IMSI: " << imsi
-                  << " Generated session " << session_id;
+                  << " PDU id " << pdu_id;
 
       /* Message may be intial or modification message. Only taken care
        * intial message. Check if it's initial message
        */
-      if ((cfg.rat_specific_context.m5gsm_session_context().rquest_type() ==
+      if ((cfg.rat_specific_context.m5gsm_session_context().request_type() ==
            INITIAL_REQUEST) &&
           (cfg.common_context.sm_session_state() == CREATING_0)) {
         /* it's new UE establisment request and need to create the session
          * context
          */
-        MLOG(MINFO)
-            << "AMF request type INITIAL_REQUEST and session state CREATING";
         auto session_map = session_store_.read_sessions({imsi});
-        send_create_session(session_map, imsi, session_id, cfg, dnn);
+        send_create_session(session_map, imsi, cfg, pdu_id);
+        response_callback(Status::OK, SmContextVoid());
+        return;
       }
+      MLOG(MERROR)
+          << "AMF request type- Unhandled request type:"
+          << cfg.rat_specific_context.m5gsm_session_context().request_type();
+
+      Status status(grpc::UNKNOWN, "Unknown session state or request");
+      response_callback(status, SmContextVoid());
+    }
+    return;

```

Comment:
`return;` seems to be a no op here, let's just remove it?

Reply:
removed the return; as there is no operation here.